From c2e7fd56f298b359d5922e2507e247f437dad76f Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:39:33 -0500
Subject: [drm] i915: resume MST after reading back hw state

Message-id: <1418852380-13061-83-git-send-email-rclark@redhat.com>
Patchwork-id: 102219
O-Subject: [RHEL7 drm 82/89] drm/i915: resume MST after reading back hw state
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: drm-next

commit e7d6f7d708290da1b7c92f533444b042c79412e0

Author:     Dave Airlie <airlied@redhat.com>
AuthorDate: Mon Dec 8 13:23:37 2014 +1000
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Mon Dec 15 14:46:53 2014 -0500

    drm/i915: resume MST after reading back hw state

    Otherwise the MST resume paths can hit DPMS paths
    which hit state checker paths, which hit WARN_ON,
    because the state checker is inconsistent with the
    hw.

    This fixes a bunch of WARN_ON's on resume after
    undocking.

    Signed-off-by: Dave Airlie <airlied@redhat.com>
    Reviewed-by: Daniel Vetter <daniel.vetter@ffwll.ch>
    Cc: stable@vger.kernel.org
    Signed-off-by: Jani Nikula <jani.nikula@intel.com>

Signed-off-by: Rob Clark <rclark@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/i915/i915_drv.c b/drivers/gpu/drm/i915/i915_drv.c
index d323da8..cdb9287 100644
--- a/drivers/gpu/drm/i915/i915_drv.c
+++ b/drivers/gpu/drm/i915/i915_drv.c
@@ -652,11 +652,12 @@ static int __i915_drm_thaw(struct drm_device *dev, bool restore_gtt_mappings)
    spin_unlock_irqrestore(&dev_priv->irq_lock, irqflags);
   }
 
-  intel_dp_mst_resume(dev);
   drm_modeset_lock_all(dev);
   intel_modeset_setup_hw_state(dev, true);
   drm_modeset_unlock_all(dev);
 
+  intel_dp_mst_resume(dev);
+
   /*
    * ... but also need to make sure that hotplug processing
    * doesn't cause havoc. Like in the driver load code we don't
-- 
1.7.1